db/migrate/20211110153757_create_user_counters.rb

```diff
@@ -0,0 +1,13 @@
+class CreateUserCounters < ActiveRecord::Migration[6.0]
+  def change
+    create_table :user_counters do |t|
+      t.integer :user_id
+      t.string :counter_name
```

Comment:
should there be a unique constraint on the combination of these 2 fields? I see you are using `first_or_initialize` when accessing a counter, but it might be beneficial to enforce the constraint at the db level.

Reply:
Sure, that's a good idea